src/main.rs

```diff
@@ -1202,115 +1203,113 @@ fn cleanup_autogenerated_file(path: &Path) {
-//                 try:
-//                     output_file = codecs.open(fullpath, "w", encoding="utf-8")
-//                 except IOError as err:
-//                     add_diagnostic("Unable to open file '%s': error(%d): %s" % (fullpath, err.errno, err.strerror))
-//                     return False
-//
-//             built_command_output.insert(0, "# " + CMDNAME)
-//
-//             # Output the magic word Autogenerated so we can tell if we can overwrite this
-//             built_command_output.insert(1, "# Autogenerated from man page " + manpage_path)
-//             # built_command_output.insert(2, "# using " + parser.__class__.__name__) # XXX MISATTRIBUTES THE CULPABILE PARSER! Was really using Type2 but reporting TypeDeroffManParser
-//
-//             for line in built_command_output:
-//                 output_file.write(line)
-//                 output_file.write('\n')
-//             output_file.write('\n')
-//             add_diagnostic(manpage_path + ' parsed successfully')
-//             if output_file != sys.stdout:
-//                 output_file.close()
-//         else:
-//             parser_names =  ', '.join(p.__class__.__name__ for p in parsersToTry)
-//             #add_diagnostic('%s contains no options or is unparsable' % manpage_path, BRIEF_VERBOSE)
-//             add_diagnostic('%s contains no options or is unparsable (tried parser %s)' % (manpage_path, parser_names), BRIEF_VERBOSE)
-//
-//     return success
+fn parse_manpage_at_path(
+    manpage_path: &Path,
+    output_directory: &Path,
+    deroff_only: bool,
+    write_to_stdout: bool,
+) -> io::Result<bool> {
+    // Clear diagnostic
+    // diagnostic_output[:] = []
+    // diagnostic_indent = 0
+
+    // Set up some diagnostic
+    // add_diagnostic(format!("Considering {}", manpage_path));
+    // diagnostic_indent += 1
+
+    let mut manpage = String::new();
+    if manpage_path.ends_with(".gz") {
+        let mut gz = GzDecoder::new(File::open(manpage_path)?);
+        gz.read_to_string(&mut manpage)?;
+    } else if manpage_path.ends_with(".bz2") {
+        let mut bz = BzDecoder::new(File::open(manpage_path)?);
+        bz.read_to_string(&mut manpage)?;
+    } else if manpage_path.ends_with(".xz") || manpage_path.ends_with(".lzma") {
+        let mut xz = XzDecoder::new(File::open(manpage_path)?);
+        xz.read_to_string(&mut manpage)?;
+    } else if [".1", ".2", ".3", ".4", ".5", ".6", ".7", ".8", ".9"]
+        .iter()
+        .any(|suffix| manpage_path.ends_with(suffix))
+    {
+        File::open(manpage_path)?.read_to_string(&mut manpage)?;
+    }
+
+    // Get the "base" command, e.g. gcc.1.gz -> gcc
+    // These casts are safe as OsStr is internally a wrapper around [u8] on all
+    // platforms. Taken from libstd.
+    let cmdname = manpage_path
+        .file_name()
+        .and_then(|file| file.as_bytes().splitn(2, |b| *b == b'.').next());
+    let cmdname = String::from_utf8_lossy(cmdname.unwrap());
+    let ignored_commands = [
```

Comment:
```rust
if [
    "cc", "g++", "gcc",
    "cpp", "emacs", "gprof",
    "wget", "ld", "awk"
].contains(&cmdname.as_ref()) {
    return Ok(false);
}
```

Reply:
I think the current one make it much easier to know those variables are part of ignored commands, the version you show does not indicate that.

Reply:
Fair